server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java

```diff
@@ -73,26 +86,56 @@ protected RowRecord nextWithoutConstraint() throws IOException {
    * @return if there has next row record.
    */
   private boolean cacheRowRecord() throws IOException {
-    while (timeGenerator.hasNext()) {
-      boolean hasField = false;
-      long timestamp = timeGenerator.next();
-      RowRecord rowRecord = new RowRecord(timestamp);
-      for (int i = 0; i < seriesReaderByTimestampList.size(); i++) {
-        IReaderByTimestamp reader = seriesReaderByTimestampList.get(i);
-        Object value = reader.getValueInTimestamp(timestamp);
-        if (value == null) {
-          rowRecord.addField(null);
-        } else {
-          hasField = true;
-          rowRecord.addField(value, dataTypes.get(i));
+    int seriesNum = seriesReaderByTimestampList.size();
+    while (timeGenerator.hasNextTimeColumn()) {
+      final TimeColumn timeColumn = timeGenerator.nextTimeColumn();
+
+      Future<List<Field>>[] futures = new Future[seriesNum];
+      for (int i = 0; i < seriesNum; i++) {
+        final IReaderByTimestamp readerByTimestamp = seriesReaderByTimestampList.get(i);
+        final TSDataType tsDataType = dataTypes.get(i);
+        futures[i] = QueryTaskPoolManager.getInstance().submit(() -> {
```

Comment:
remove parallelization, do this in a separate PR and do some experiments to help us decide whether introduce parallelization

Reply:
done